File: 209. Minimum Size Subarray Sum/main.py
import sys
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        result = sys.maxsize
        sum = 0
        left = 0

        for i in range(len(nums)):
            sum += nums[i]

            while sum >= target:
                result = min(result, i - left + 1)
                sum -= nums[left]
                left += 1

        if result == sys.maxsize:
            return 0

        return result


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        result = sys.maxsize
        sum = 0
        left = 0

        for i, num in enumerate(nums):
            sum += num

            while sum >= target:
                result = min(result, i - left + 1)
                sum -= nums[left]
                left += 1

        if result == sys.maxsize:
            return 0

        return result

File: 209. Minimum Size Subarray Sum/main.rs
struct Solution;

impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let mut result = i32::MAX;
        let mut sum = 0;
        let mut left = 0;

        for (i, num) in nums.iter().enumerate() {
            sum += num;

            while sum >= target {
                result = result.min(i as i32 - left + 1);
                sum -= nums[left as usize];
                left += 1;
            }
        }

        if result == i32::MAX {
            0
        } else {
            result
        }
    }
}

impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let mut result = i32::MAX;
        let mut sum = 0;
        let mut left = 0;

        for i in 0..nums.len() {
            sum += nums[i];

            while sum >= target {
                result = result.min(i as i32 - left + 1);
                sum -= nums[left as usize];
                left += 1;
            }
        }

        match result {
            i32::MAX => 0,
            _ => result,
        }
    }
}
